cmd/frontman/main.go

```diff
@@ -315,6 +315,13 @@ func handleFlagInputOutput(inputFile string, outputFile string, oneRunOnlyMode b
 
 func handleFlagOneRunOnlyMode(fm *frontman.Frontman, oneRunOnlyMode bool, inputFile string, output *os.File, interruptChan chan struct{}) {
 	if oneRunOnlyMode {
+		if err := fm.HealthCheck(); err != nil {
+			log.WithError(err).Warningln("Health checks are not passed. Skipping other checks.")
+			return
+		} else if fm.HealthCheckNotPassed {
```

Comment:
This check happens only if `fm.HealthCheck()` returned `nil` on the current iteration, that means check has been failing previously, but now it is fixed. In that case we issue a warning that tells about clearance.

I won't refactor my if-elses anymore, except there is an explicit error. Because such refactoring may introduce real bugs and has no real profit itself.

Reply:
> I won't refactor my if-elses anymore, except there is an explicit error. Because such refactoring may introduce real bugs and has no real profit itself.

It's a not necessary level of conditions. It makes the code simpler and easier to read. Why should this introduce real bugs? I personally think simpler code _is_ a profit on itself.

Reply:
@soupdiver, I dropped the else in d632707c8f3c32264dcb7d8a70a434283eadc734